test_bandwidth.py

```diff
@@ -0,0 +1,204 @@
+import os
+import sys
+import time
+import argparse
+from abc import abstractmethod
+
+import torch
+import torch.distributed as dist
+from typing import List
+from prettytable import PrettyTable
+os.environ['CUDA_VISIBLE_DEVICES'] = '1,2,3,4'
+    
+def setup(rank, world_size):
+    os.environ["MASTER_ADDR"] = os.getenv("MASTER_ADDR", "localhost")
+    os.environ["MASTER_PORT"] = os.getenv("MASTER_PORT", "12355")
+    os.environ["RANK"] = os.getenv("RANK", "0")
+    os.environ["WORLD_SIZE"] = os.getenv("WORLD_SIZE", "1")
+    dist.init_process_group(world_size=world_size, rank=rank,
```

Comment:
You can two sources for rank and world_size:
1. from the function argument
2. from environment

what's more is that the function argument is also fetched from environment variable, this makes this code snippet rather messy and error-prone.

Reply:
Sorry, I'll remove the args of the function argument.